refinance/config.py

```diff
@@ -1,21 +1,41 @@
 """Application configuration"""
 
-from dataclasses import dataclass
+from logging import getLogger
+from os import environ
+from typing import Any, Self
 
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+logger = getLogger(__name__)
+
+
+class Config(BaseSettings):
+    model_config = SettingsConfigDict(env_file_encoding="utf-8", env_nested_delimiter="__", extra="ignore")
 
-@dataclass
-class Config:
     app_name: str = "refinance"
     app_version: str = "0.1.0"
 
     @property
-    def database_path(self):
+    def database_path(self) -> str:
         return f"./{self.app_name}.db"
 
     @property
-    def database_url(self):
+    def database_url(self) -> str:
         return f"sqlite:///{self.database_path}"
 
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize config."""
+        super().__init__(*args, **kwargs)
+        logger.debug(f"Config initialized: {self.model_dump()}")
+
+    @classmethod
+    def from_env(cls) -> Self:
```

Comment:
What's the use case of `from_env` classmethod there?

Reply:
Initialization of Config class by values from environment variables and secret files.

If you just call `__init__` then initialization will be only by environment variables.

Reply:
Storing secrets in a directory rather than as envvars override looks like additional complexity.